client/profile-wizard/index.js

```diff
@@ -223,6 +232,29 @@ class ProfileWizard extends Component {
 		}
 	}
 
+	skipProfiler() {
+		const { createNotice, updateProfileItems } = this.props;
+		updateProfileItems( {
+			skipped: true,
+			step: this.getCurrentStep().key,
+		} )
+			.then( ( response ) => {
+				if ( response.status === 'success' ) {
+					recordEvent( 'storeprofiler_store_details_skip' );
+					getHistory().push( getNewPath( {}, '/', {} ) );
+				}
```

Comment:
We won't need to check `response.status` here since this is already checked in the data actions.  The `.then()` block will only be triggered if the response was successful:

```suggestion
				recordEvent( 'storeprofiler_store_details_skip' );
				getHistory().push( getNewPath( {}, '/', {} ) );
```

Reply:
> We can probably safely remove any reference to this saved step and "lastStep since it will always be the first step if they skip.

Now I see, thank you @joshuatf, I removed it in the commit 37cab6

> We won't need to check response.status here since this is already checked in the data actions. The .then() block will only be triggered if the response was successful:

It seems like there is something tricky going on here, because with some types of errors the response will be `undefined` but in a "success" situation. So the user would be redirected to the home screen without receiving an alert.
The `catch` is triggered because the `response` doesn't have a `status`.
I created the branch `add/4568_test_2` with [the line commented](https://github.com/woocommerce/woocommerce-admin/pull/4721/files#diff-713853358fa08698c1bf0f8dfa42ca74R234) and adding a `die()` in the [update_items method](https://github.com/woocommerce/woocommerce-admin/blob/add/4568_test_2/src/API/OnboardingProfile.php#L129) if you want to test it.

Reply:
Ah, I see.  Good catch.  So after this line:
https://github.com/woocommerce/woocommerce-admin/blob/bd0d7dce14ebc3809f1d1bacd0155c07dc6345db/packages/data/src/onboarding/actions.js#L56

We're also missing: `throw new Error();` which will allow us to catch errors after `updateProfileItems()`.  Do you mind adding that here?  It seems to get things working for me with your demo branch, but  we should definitely check and make sure it doesn't break other areas of the onboarding.

<img width="473" alt="Screen Shot 2020-07-15 at 6 39 42 PM" src="https://user-images.githubusercontent.com/10561050/87565668-9db80e00-c6ca-11ea-8e93-7af69858931f.png">

Reply:
Got it! I added the code in the commit 3705935
I did a rebase and updated the test branch (`add/4568_test`). Now I'm testing and it looks like everything is working correctly now.
Could you take another look at this PR?